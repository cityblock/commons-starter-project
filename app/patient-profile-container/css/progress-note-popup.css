@import '../../shared/css/sizes';

.tabContainer {
  display: block;
}

.topBar {
  width: 100%;
  background: var(--blue-color);
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 10px;
}

.topBarLabel {
  color: var(--light-blue-color);
  font-family: var(--basetica-medium);
  font-size: var(--default-font-size);
}

.closeButton {
  background: url('../../assets/chevron-down-white.svg');
  background-repeat: no-repeat;
  background-position: center center;
  padding: 10px;
  width: 15px;
  height: 15px;
  cursor: pointer;
  opacity: 1;
  transition: opacity 0.6s ease-in-out;
}

.closeButton:hover {
  transition: opacity 0.3s ease-in-out;
  opacity: 0.8;
}

.middleBar {
  display: flex;
  justify-content: space-between;
  padding: 16px 20px;
  background: var(--white-font-color);
}

.button {
  @apply --button;

  height: 32px;
  padding: 8px 33px 7px;
}

.buttonInactive {
  opacity: 0.5;
}

.encounterTypeContainer {
  background: var(--white-font-color);
  padding: 20px;
}

.encounterTypeLabel {
  font-family: var(--basetica-medium);
  font-size: var(--default-font-size);
}

.encounterTypeSelect {
  @apply --select;
}

.locationTimeRow {
  padding-top: var(--small-gutter);
  display: flex;
}

.location {
  flex: 1;
  padding-right: var(--small-gutter);
}

.time {
  flex: 1;
}

.patientContainer {
  display: block;
}

.patientPhoto {
  background-repeat: no-repeat;
  background-size: cover;
  background-position: center center;
  width: 43px;
  height: 43px;
  display: inline-block;
  vertical-align: top;
}

.patientContainerRight {
  margin-left: 20px;
  display: inline-block;
  vertical-align: top;
}

.patientName {
  font-weight: var(--roboto-medium-weight);
  font-size: var(--medium-font-size);
}

.openedAt {
  display: block;
}

.error {
  color: var(--red-color);
}
