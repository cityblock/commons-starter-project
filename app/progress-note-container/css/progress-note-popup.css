@import '../../shared/css/sizes';

.container {
  position: fixed;
  bottom: 0;
  right: 0;
  height: 0;
  width: 550px;
  background: var(--white-color);
  z-index: 11;
  transition: height 0.6s ease-in-out;
  box-shadow: var(--login-drop-shadow);
}

.visible {
  transition: height 0.3s ease-in-out;
  height: calc(100vh - var(--header-height));
}

.bottomSection {
  overflow: auto;
  height: calc(100vh - 235px);
}

.tabContainer {
  display: block;
}

.topBar {
  width: 100%;
  background: var(--navy-color);
  height: 50px;
  display: flex;
  justify-content: space-between;
  align-items: center;
  font-size: var(--medium-font-size);
  padding: 26px var(--small-gutter) 23px;
}

.topBarLabel {
  color: var(--light-blue-color);
  font-family: var(--basetica-medium);
  font-size: var(--default-font-size);
}

.closeButton {
  background: url('../../assets/chevron-down-white.svg');
  background-repeat: no-repeat;
  background-position: center center;
  padding: 10px;
  width: 15px;
  height: 15px;
  cursor: pointer;
  opacity: 1;
  transition: opacity 0.6s ease-in-out;
}

.closeButton:hover {
  transition: opacity 0.3s ease-in-out;
  opacity: 0.8;
}

.middleBar {
  display: flex;
  justify-content: space-between;
  padding: 16px var(--small-gutter);
  background: var(--white-font-color);
}

.button {
  @apply --button;

  height: 32px;
  padding: 8px 33px 7px;
}

.buttonInactive {
  opacity: 0.5;
}

.patientContainer {
  display: block;
}

.patientPhoto {
  background-repeat: no-repeat;
  background-size: cover;
  background-position: center center;
  width: 43px;
  height: 43px;
  display: inline-block;
  vertical-align: top;
}

.patientContainerRight {
  margin-left: var(--small-gutter);
  display: inline-block;
  vertical-align: top;
}

.patientName {
  font-weight: var(--roboto-medium-weight);
  font-size: var(--medium-font-size);
}

.openedAt {
  display: block;
}
