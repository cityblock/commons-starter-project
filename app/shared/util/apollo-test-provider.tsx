import { InMemoryCache } from 'apollo-cache-inmemory';
import { ApolloClient } from 'apollo-client';
import { SchemaLink } from 'apollo-link-schema';
import { print } from 'graphql';
import { addMockFunctionsToSchema, makeExecutableSchema } from 'graphql-tools';
import createHistory from 'history/createBrowserHistory';
import * as React from 'react';
import { ApolloProvider } from 'react-apollo';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import * as schema from '../../../server/graphql/schema.graphql';
import ReduxConnectedIntlProvider from '../../redux-connected-intl-provider';
import createStore from '../../store';

const history = createHistory();
const store = createStore(history);

interface IProps {
  graphqlMocks: any;
}

// Helper in our codebase to create the apollo client
// from http://blog.dideric.is/2018/03/18/Testing-apollo-containers/
export default class ApolloTestProvider extends React.Component<IProps> {
  static createSchema = () => makeExecutableSchema({ typeDefs: print(schema) });

  apolloClient: any;
  schema: any;

  constructor(props: IProps) {
    super(props);
    // Every instance should have it's own schema instance so tests
    // don't bleed into one another
    this.schema = ApolloTestProvider.createSchema();
    this.apolloClient = new ApolloClient<any>({
      link: new SchemaLink({ schema: this.schema }),
      cache: new InMemoryCache().restore({}) as any,
    });
    this.addDefaultMocks();
  }

  componentWillMount() {
    const { graphqlMocks } = this.props;
    this.mockGraphql(graphqlMocks);
  }

  componentWillReceiveProps({ graphqlMocks }: any) {
    this.mockGraphql(graphqlMocks);
  }

  addDefaultMocks() {
    addMockFunctionsToSchema({
      schema: this.schema,
    });
  }

  mockGraphql(mocks = {}) {
    addMockFunctionsToSchema({
      schema: this.schema,
      mocks,
    });
  }

  render() {
    const { children } = this.props;
    return (
      <Provider store={store}>
        <ApolloProvider client={this.apolloClient}>
          <ReduxConnectedIntlProvider defaultLocale="en">
            <MemoryRouter>{children}</MemoryRouter>
          </ReduxConnectedIntlProvider>
        </ApolloProvider>
      </Provider>
    );
  }
}
