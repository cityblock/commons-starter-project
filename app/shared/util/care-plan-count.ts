import { FullCarePlanSuggestionFragment } from '../../graphql/types';

export function getConcernCount(carePlanSuggestions: FullCarePlanSuggestionFragment[]) {
  if (!carePlanSuggestions.length) {
    return 0;
  }

  const concernSuggestions = carePlanSuggestions.filter(
    suggestion => suggestion.suggestionType === 'concern',
  );

  return concernSuggestions.length;
}

export function getGoalSuggestions(carePlanSuggestions: FullCarePlanSuggestionFragment[]) {
  if (!carePlanSuggestions.length) {
    return [];
  }

  return carePlanSuggestions.filter(suggestion => suggestion.suggestionType === 'goal');
}

export function getGoalCount(carePlanSuggestions: FullCarePlanSuggestionFragment[]) {
  return getGoalSuggestions(carePlanSuggestions).length;
}

export function getTaskCount(carePlanSuggestions: FullCarePlanSuggestionFragment[]) {
  if (getGoalCount(carePlanSuggestions) === 0) {
    return 0;
  }

  const goalSuggestions = getGoalSuggestions(carePlanSuggestions);

  const taskSuggestions = goalSuggestions
    .map(goalSuggestion => goalSuggestion.goalSuggestionTemplate!.taskTemplates)
    .reduce((taskSuggestions1, taskSuggestions2) => taskSuggestions1.concat(taskSuggestions2));

  return (taskSuggestions || []).length;
}
