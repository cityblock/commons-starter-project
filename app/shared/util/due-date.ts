export const checkIfDueSoon = (dateString: string | null) => {
  let dueSoon = false;

  if (dateString) {
    const taskDueDate = new Date(dateString);
    dueSoon = taskDueDate.valueOf() < Date.now() + 60 * 60 * 24 * 1000;
  }

  return dueSoon;
};
