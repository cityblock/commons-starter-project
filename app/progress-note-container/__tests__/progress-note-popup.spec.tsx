import { shallow } from 'enzyme';
import * as React from 'react';
import { patient, progressNote, progressNoteTemplate } from '../../shared/util/test-data';
import { ProgressNotePopup as Component } from '../progress-note-popup';

const oldDate = Date.now;
const editProgressNote = jest.fn();
const completeProgressNote = jest.fn();
const getOrCreateProgressNote = jest.fn();
const progressNoteTemplates = [progressNoteTemplate];

describe('builder concerns', () => {
  beforeAll(() => {
    Date.now = jest.fn(() => 1500494779252);
  });
  afterAll(() => {
    Date.now = oldDate;
  });

  describe('renders progress note popup with templates', () => {
    it('renders progress note popup with templates', async () => {
      const component = shallow(
        <Component
          close={jest.fn()}
          patientId={patient.id}
          visible={true}
          patient={patient}
          editProgressNote={editProgressNote}
          completeProgressNote={completeProgressNote}
          createProgressNote={getOrCreateProgressNote}
          progressNoteTemplates={progressNoteTemplates}
        />,
      );
      const instance = component.instance() as Component;
      const result = instance.render();
      expect(result).toMatchSnapshot();
    });
    it('renders progress note', async () => {
      const component = shallow(
        <Component
          close={jest.fn()}
          patientId={patient.id}
          visible={true}
          patient={patient}
          editProgressNote={editProgressNote}
          completeProgressNote={completeProgressNote}
          createProgressNote={getOrCreateProgressNote}
          progressNote={progressNote}
          progressNoteTemplates={progressNoteTemplates}
        />,
      );
      const instance = component.instance() as Component;

      const result = instance.render();
      expect(result).toMatchSnapshot();
    });
  });
});
