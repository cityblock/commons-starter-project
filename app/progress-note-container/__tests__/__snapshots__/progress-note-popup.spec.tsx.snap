// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`builder concerns renders progress note popup with templates renders progress note 1`] = `
<div>
  <div
    className="topBar"
  >
    <div
      className="topBarLabel"
    >
      Progress Note: 
      Progress Note Template Title
    </div>
    <div
      className="closeButton"
      onClick={[MockFunction]}
    />
  </div>
  <div
    className="middleBar"
  >
    <Link
      className="patientContainer"
      replace={false}
      to="/patients/patient-id"
    >
      <Apollo(PatientPhoto)
        gender="male"
        hasUploadedPhoto={false}
        patientId="patient-id"
        type="circleLarge"
      />
      <div
        className="patientContainerRight"
      >
        <div
          className="patientName"
        >
          Bob Smith
        </div>
        <FormattedRelative
          updateInterval={10000}
          value="2017-09-07T13:45:14.532Z"
        >
          [Function]
        </FormattedRelative>
      </div>
    </Link>
    <Button
      color="blue"
      disabled={true}
      messageId="patient.submitProgressNote"
      onClick={[Function]}
    />
  </div>
  <UnderlineTabs>
    <UnderlineTab
      messageId="patient.context"
      onClick={[Function]}
      selected={true}
    />
    <UnderlineTab
      messageId="patient.activity"
      onClick={[Function]}
      selected={false}
    />
  </UnderlineTabs>
  <div
    className="bottomSection"
  >
    <Apollo(ProgressNoteContext)
      close={[MockFunction]}
      disabled={false}
      onChange={[Function]}
      progressNote={
        Object {
          "completedAt": null,
          "createdAt": "2017-09-07T13:45:14.532Z",
          "deletedAt": null,
          "id": "progress-note-id",
          "location": null,
          "memberConcern": "member concern",
          "needsSupervisorReview": null,
          "patient": Object {
            "cityblockId": 123,
            "computedPatientStatus": Object {
              "isAdvancedDirectivesAdded": true,
              "isConsentSigned": false,
              "isCoreIdentityVerified": true,
              "isDemographicInfoUpdated": true,
              "isEmergencyContactAdded": false,
              "isPhotoAddedOrDeclined": false,
            },
            "coreIdentityVerifiedAt": "2017-09-07T13:45:14.532Z",
            "coreIdentityVerifiedById": "id2",
            "createdAt": "2017-09-07T13:45:14.532Z",
            "dateOfBirth": "01/01/1999",
            "deletedAt": null,
            "firstName": "Bob",
            "homeClinicId": "clinic-id",
            "id": "patient-id",
            "lastName": "Smith",
            "middleName": null,
            "mrn": "12345",
            "nmi": "12345",
            "patientDataFlags": Array [],
            "patientInfo": Object {
              "canReceiveCalls": true,
              "gender": "male",
              "genderFreeText": null,
              "googleCalendarId": null,
              "hasDeclinedPhotoUpload": false,
              "hasEmail": true,
              "hasHealthcareProxy": false,
              "hasMolst": false,
              "hasUploadedPhoto": false,
              "id": "patient-info-id",
              "isMarginallyHoused": false,
              "language": "en",
              "maritalStatus": "currentlyMarried",
              "preferredContactMethod": "phone",
              "preferredContactTime": "afternoon",
              "preferredName": "Bobby",
              "primaryAddress": Object {
                "city": "Brooklyn",
                "description": "work",
                "id": "address-1-id",
                "state": "NY",
                "street1": "55 Washington St",
                "street2": "Unit 552",
                "zip": "11201",
              },
              "primaryEmail": Object {
                "emailAddress": "test@email.com",
                "id": "email-1-id",
              },
              "primaryPhone": Object {
                "description": "main phone number",
                "id": "phone-1-id",
                "phoneNumber": "555-555-5555",
                "type": "mobile",
              },
              "transgender": "no",
            },
            "patientState": Object {
              "currentState": "outreach",
              "id": "patientStateId",
            },
            "scratchPad": "Note",
            "ssnEnd": "1234",
            "updatedAt": "2017-09-07T13:45:14.532Z",
          },
          "patientId": "patient-id",
          "progressNoteTemplate": Object {
            "createdAt": "2017-09-07T13:45:14.532Z",
            "deletedAt": null,
            "id": "progress-note-template-id",
            "title": "Progress Note Template Title",
            "updatedAt": "2017-09-07T13:45:14.532Z",
          },
          "progressNoteTemplateId": "progress-note-template-id",
          "reviewedBySupervisorAt": null,
          "startedAt": null,
          "summary": "summary!",
          "supervisor": null,
          "supervisorId": null,
          "supervisorNotes": null,
          "updatedAt": "2017-09-07T13:45:14.532Z",
          "user": Object {
            "awayMessage": "Winter is here",
            "createdAt": "2017-09-07T13:45:14.532Z",
            "email": "a@b.com",
            "firstName": "first",
            "googleProfileImageUrl": null,
            "homeClinicId": "clinic-id",
            "id": "id",
            "isAvailable": true,
            "lastName": "last",
            "locale": "en",
            "permissions": "orange",
            "phone": "(212) 555-8394",
            "updatedAt": "2017-09-07T13:45:14.532Z",
            "userRole": "physician",
          },
          "worryScore": 2,
        }
      }
      progressNoteTemplates={
        Array [
          Object {
            "createdAt": "2017-09-07T13:45:14.532Z",
            "deletedAt": null,
            "id": "progress-note-template-id",
            "title": "Progress Note Template Title",
            "updatedAt": "2017-09-07T13:45:14.532Z",
          },
        ]
      }
      questions={
        Array [
          Object {
            "answerType": "radio",
            "answers": Array [
              Object {
                "concernSuggestions": Array [],
                "createdAt": "Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)",
                "deletedAt": null,
                "displayValue": "answer value",
                "goalSuggestions": Array [],
                "id": "answer-id",
                "inSummary": true,
                "order": 1,
                "questionId": "cool-task-id",
                "riskAdjustmentType": "increment",
                "riskArea": null,
                "screeningTool": null,
                "summaryText": "summary text",
                "updatedAt": "Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)",
                "value": "true",
                "valueType": "boolean",
              },
            ],
            "applicableIfQuestionConditions": Array [
              Object {
                "answerId": "answer-id",
                "id": "question-condition",
                "questionId": "cool-question-id",
              },
            ],
            "applicableIfType": "allTrue",
            "computedField": null,
            "computedFieldId": null,
            "createdAt": "Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)",
            "deletedAt": null,
            "id": "cool-question-id",
            "order": 1,
            "otherTextAnswerId": null,
            "riskAreaId": "risk-area-id",
            "screeningToolId": null,
            "title": "Question Title",
            "updatedAt": "Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)",
            "validatedSource": "validated source",
          },
        ]
      }
    />
    <Apollo(Apollo(ProgressNoteCosignature))
      disabled={false}
      progressNote={
        Object {
          "completedAt": null,
          "createdAt": "2017-09-07T13:45:14.532Z",
          "deletedAt": null,
          "id": "progress-note-id",
          "location": null,
          "memberConcern": "member concern",
          "needsSupervisorReview": null,
          "patient": Object {
            "cityblockId": 123,
            "computedPatientStatus": Object {
              "isAdvancedDirectivesAdded": true,
              "isConsentSigned": false,
              "isCoreIdentityVerified": true,
              "isDemographicInfoUpdated": true,
              "isEmergencyContactAdded": false,
              "isPhotoAddedOrDeclined": false,
            },
            "coreIdentityVerifiedAt": "2017-09-07T13:45:14.532Z",
            "coreIdentityVerifiedById": "id2",
            "createdAt": "2017-09-07T13:45:14.532Z",
            "dateOfBirth": "01/01/1999",
            "deletedAt": null,
            "firstName": "Bob",
            "homeClinicId": "clinic-id",
            "id": "patient-id",
            "lastName": "Smith",
            "middleName": null,
            "mrn": "12345",
            "nmi": "12345",
            "patientDataFlags": Array [],
            "patientInfo": Object {
              "canReceiveCalls": true,
              "gender": "male",
              "genderFreeText": null,
              "googleCalendarId": null,
              "hasDeclinedPhotoUpload": false,
              "hasEmail": true,
              "hasHealthcareProxy": false,
              "hasMolst": false,
              "hasUploadedPhoto": false,
              "id": "patient-info-id",
              "isMarginallyHoused": false,
              "language": "en",
              "maritalStatus": "currentlyMarried",
              "preferredContactMethod": "phone",
              "preferredContactTime": "afternoon",
              "preferredName": "Bobby",
              "primaryAddress": Object {
                "city": "Brooklyn",
                "description": "work",
                "id": "address-1-id",
                "state": "NY",
                "street1": "55 Washington St",
                "street2": "Unit 552",
                "zip": "11201",
              },
              "primaryEmail": Object {
                "emailAddress": "test@email.com",
                "id": "email-1-id",
              },
              "primaryPhone": Object {
                "description": "main phone number",
                "id": "phone-1-id",
                "phoneNumber": "555-555-5555",
                "type": "mobile",
              },
              "transgender": "no",
            },
            "patientState": Object {
              "currentState": "outreach",
              "id": "patientStateId",
            },
            "scratchPad": "Note",
            "ssnEnd": "1234",
            "updatedAt": "2017-09-07T13:45:14.532Z",
          },
          "patientId": "patient-id",
          "progressNoteTemplate": Object {
            "createdAt": "2017-09-07T13:45:14.532Z",
            "deletedAt": null,
            "id": "progress-note-template-id",
            "title": "Progress Note Template Title",
            "updatedAt": "2017-09-07T13:45:14.532Z",
          },
          "progressNoteTemplateId": "progress-note-template-id",
          "reviewedBySupervisorAt": null,
          "startedAt": null,
          "summary": "summary!",
          "supervisor": null,
          "supervisorId": null,
          "supervisorNotes": null,
          "updatedAt": "2017-09-07T13:45:14.532Z",
          "user": Object {
            "awayMessage": "Winter is here",
            "createdAt": "2017-09-07T13:45:14.532Z",
            "email": "a@b.com",
            "firstName": "first",
            "googleProfileImageUrl": null,
            "homeClinicId": "clinic-id",
            "id": "id",
            "isAvailable": true,
            "lastName": "last",
            "locale": "en",
            "permissions": "orange",
            "phone": "(212) 555-8394",
            "updatedAt": "2017-09-07T13:45:14.532Z",
            "userRole": "physician",
          },
          "worryScore": 2,
        }
      }
      updateProgressNote={[Function]}
    />
  </div>
</div>
`;

exports[`builder concerns renders progress note popup with templates renders progress note popup with templates 1`] = `
<div>
  <div
    className="topBar"
  >
    <div
      className="topBarLabel"
    >
      Progress Note: 
      Progress Note Template Title
    </div>
    <div
      className="closeButton"
      onClick={[MockFunction]}
    />
  </div>
  <div
    className="middleBar"
  >
    <Link
      className="patientContainer"
      replace={false}
      to="/patients/patient-id"
    >
      <Apollo(PatientPhoto)
        gender="male"
        hasUploadedPhoto={false}
        patientId="patient-id"
        type="circleLarge"
      />
      <div
        className="patientContainerRight"
      >
        <div
          className="patientName"
        >
          Bob Smith
        </div>
        <FormattedRelative
          updateInterval={10000}
          value="2017-09-07T13:45:14.532Z"
        >
          [Function]
        </FormattedRelative>
      </div>
    </Link>
    <Button
      color="blue"
      disabled={true}
      messageId="patient.submitProgressNote"
      onClick={[Function]}
    />
  </div>
  <UnderlineTabs>
    <UnderlineTab
      messageId="patient.context"
      onClick={[Function]}
      selected={true}
    />
    <UnderlineTab
      messageId="patient.activity"
      onClick={[Function]}
      selected={false}
    />
  </UnderlineTabs>
  <div
    className="bottomSection"
  >
    <Apollo(ProgressNoteContext)
      close={[MockFunction]}
      disabled={false}
      onChange={[Function]}
      progressNote={
        Object {
          "completedAt": null,
          "createdAt": "2017-09-07T13:45:14.532Z",
          "deletedAt": null,
          "id": "progress-note-id",
          "location": null,
          "memberConcern": "member concern",
          "needsSupervisorReview": null,
          "patient": Object {
            "cityblockId": 123,
            "computedPatientStatus": Object {
              "isAdvancedDirectivesAdded": true,
              "isConsentSigned": false,
              "isCoreIdentityVerified": true,
              "isDemographicInfoUpdated": true,
              "isEmergencyContactAdded": false,
              "isPhotoAddedOrDeclined": false,
            },
            "coreIdentityVerifiedAt": "2017-09-07T13:45:14.532Z",
            "coreIdentityVerifiedById": "id2",
            "createdAt": "2017-09-07T13:45:14.532Z",
            "dateOfBirth": "01/01/1999",
            "deletedAt": null,
            "firstName": "Bob",
            "homeClinicId": "clinic-id",
            "id": "patient-id",
            "lastName": "Smith",
            "middleName": null,
            "mrn": "12345",
            "nmi": "12345",
            "patientDataFlags": Array [],
            "patientInfo": Object {
              "canReceiveCalls": true,
              "gender": "male",
              "genderFreeText": null,
              "googleCalendarId": null,
              "hasDeclinedPhotoUpload": false,
              "hasEmail": true,
              "hasHealthcareProxy": false,
              "hasMolst": false,
              "hasUploadedPhoto": false,
              "id": "patient-info-id",
              "isMarginallyHoused": false,
              "language": "en",
              "maritalStatus": "currentlyMarried",
              "preferredContactMethod": "phone",
              "preferredContactTime": "afternoon",
              "preferredName": "Bobby",
              "primaryAddress": Object {
                "city": "Brooklyn",
                "description": "work",
                "id": "address-1-id",
                "state": "NY",
                "street1": "55 Washington St",
                "street2": "Unit 552",
                "zip": "11201",
              },
              "primaryEmail": Object {
                "emailAddress": "test@email.com",
                "id": "email-1-id",
              },
              "primaryPhone": Object {
                "description": "main phone number",
                "id": "phone-1-id",
                "phoneNumber": "555-555-5555",
                "type": "mobile",
              },
              "transgender": "no",
            },
            "patientState": Object {
              "currentState": "outreach",
              "id": "patientStateId",
            },
            "scratchPad": "Note",
            "ssnEnd": "1234",
            "updatedAt": "2017-09-07T13:45:14.532Z",
          },
          "patientId": "patient-id",
          "progressNoteTemplate": Object {
            "createdAt": "2017-09-07T13:45:14.532Z",
            "deletedAt": null,
            "id": "progress-note-template-id",
            "title": "Progress Note Template Title",
            "updatedAt": "2017-09-07T13:45:14.532Z",
          },
          "progressNoteTemplateId": "progress-note-template-id",
          "reviewedBySupervisorAt": null,
          "startedAt": null,
          "summary": "summary!",
          "supervisor": null,
          "supervisorId": null,
          "supervisorNotes": null,
          "updatedAt": "2017-09-07T13:45:14.532Z",
          "user": Object {
            "awayMessage": "Winter is here",
            "createdAt": "2017-09-07T13:45:14.532Z",
            "email": "a@b.com",
            "firstName": "first",
            "googleProfileImageUrl": null,
            "homeClinicId": "clinic-id",
            "id": "id",
            "isAvailable": true,
            "lastName": "last",
            "locale": "en",
            "permissions": "orange",
            "phone": "(212) 555-8394",
            "updatedAt": "2017-09-07T13:45:14.532Z",
            "userRole": "physician",
          },
          "worryScore": 2,
        }
      }
      progressNoteTemplates={
        Array [
          Object {
            "createdAt": "2017-09-07T13:45:14.532Z",
            "deletedAt": null,
            "id": "progress-note-template-id",
            "title": "Progress Note Template Title",
            "updatedAt": "2017-09-07T13:45:14.532Z",
          },
        ]
      }
      questions={
        Array [
          Object {
            "answerType": "radio",
            "answers": Array [
              Object {
                "concernSuggestions": Array [],
                "createdAt": "Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)",
                "deletedAt": null,
                "displayValue": "answer value",
                "goalSuggestions": Array [],
                "id": "answer-id",
                "inSummary": true,
                "order": 1,
                "questionId": "cool-task-id",
                "riskAdjustmentType": "increment",
                "riskArea": null,
                "screeningTool": null,
                "summaryText": "summary text",
                "updatedAt": "Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)",
                "value": "true",
                "valueType": "boolean",
              },
            ],
            "applicableIfQuestionConditions": Array [
              Object {
                "answerId": "answer-id",
                "id": "question-condition",
                "questionId": "cool-question-id",
              },
            ],
            "applicableIfType": "allTrue",
            "computedField": null,
            "computedFieldId": null,
            "createdAt": "Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)",
            "deletedAt": null,
            "id": "cool-question-id",
            "order": 1,
            "otherTextAnswerId": null,
            "riskAreaId": "risk-area-id",
            "screeningToolId": null,
            "title": "Question Title",
            "updatedAt": "Thu Jul 13 2017 16:52:56 GMT-0400 (EDT)",
            "validatedSource": "validated source",
          },
        ]
      }
    />
    <Apollo(Apollo(ProgressNoteCosignature))
      disabled={false}
      progressNote={
        Object {
          "completedAt": null,
          "createdAt": "2017-09-07T13:45:14.532Z",
          "deletedAt": null,
          "id": "progress-note-id",
          "location": null,
          "memberConcern": "member concern",
          "needsSupervisorReview": null,
          "patient": Object {
            "cityblockId": 123,
            "computedPatientStatus": Object {
              "isAdvancedDirectivesAdded": true,
              "isConsentSigned": false,
              "isCoreIdentityVerified": true,
              "isDemographicInfoUpdated": true,
              "isEmergencyContactAdded": false,
              "isPhotoAddedOrDeclined": false,
            },
            "coreIdentityVerifiedAt": "2017-09-07T13:45:14.532Z",
            "coreIdentityVerifiedById": "id2",
            "createdAt": "2017-09-07T13:45:14.532Z",
            "dateOfBirth": "01/01/1999",
            "deletedAt": null,
            "firstName": "Bob",
            "homeClinicId": "clinic-id",
            "id": "patient-id",
            "lastName": "Smith",
            "middleName": null,
            "mrn": "12345",
            "nmi": "12345",
            "patientDataFlags": Array [],
            "patientInfo": Object {
              "canReceiveCalls": true,
              "gender": "male",
              "genderFreeText": null,
              "googleCalendarId": null,
              "hasDeclinedPhotoUpload": false,
              "hasEmail": true,
              "hasHealthcareProxy": false,
              "hasMolst": false,
              "hasUploadedPhoto": false,
              "id": "patient-info-id",
              "isMarginallyHoused": false,
              "language": "en",
              "maritalStatus": "currentlyMarried",
              "preferredContactMethod": "phone",
              "preferredContactTime": "afternoon",
              "preferredName": "Bobby",
              "primaryAddress": Object {
                "city": "Brooklyn",
                "description": "work",
                "id": "address-1-id",
                "state": "NY",
                "street1": "55 Washington St",
                "street2": "Unit 552",
                "zip": "11201",
              },
              "primaryEmail": Object {
                "emailAddress": "test@email.com",
                "id": "email-1-id",
              },
              "primaryPhone": Object {
                "description": "main phone number",
                "id": "phone-1-id",
                "phoneNumber": "555-555-5555",
                "type": "mobile",
              },
              "transgender": "no",
            },
            "patientState": Object {
              "currentState": "outreach",
              "id": "patientStateId",
            },
            "scratchPad": "Note",
            "ssnEnd": "1234",
            "updatedAt": "2017-09-07T13:45:14.532Z",
          },
          "patientId": "patient-id",
          "progressNoteTemplate": Object {
            "createdAt": "2017-09-07T13:45:14.532Z",
            "deletedAt": null,
            "id": "progress-note-template-id",
            "title": "Progress Note Template Title",
            "updatedAt": "2017-09-07T13:45:14.532Z",
          },
          "progressNoteTemplateId": "progress-note-template-id",
          "reviewedBySupervisorAt": null,
          "startedAt": null,
          "summary": "summary!",
          "supervisor": null,
          "supervisorId": null,
          "supervisorNotes": null,
          "updatedAt": "2017-09-07T13:45:14.532Z",
          "user": Object {
            "awayMessage": "Winter is here",
            "createdAt": "2017-09-07T13:45:14.532Z",
            "email": "a@b.com",
            "firstName": "first",
            "googleProfileImageUrl": null,
            "homeClinicId": "clinic-id",
            "id": "id",
            "isAvailable": true,
            "lastName": "last",
            "locale": "en",
            "permissions": "orange",
            "phone": "(212) 555-8394",
            "updatedAt": "2017-09-07T13:45:14.532Z",
            "userRole": "physician",
          },
          "worryScore": 2,
        }
      }
      updateProgressNote={[Function]}
    />
  </div>
</div>
`;
